app/classifier/tasks/drawing/grid-buttons.cjsx

```diff
@@ -0,0 +1,188 @@
+React = require 'react'
+Select = require 'react-select'
+
+module.exports = React.createClass
+  displayName: 'GridButtons'
+
+  getDefaultProps: ->
+    preferences: null
+
+  componentWillMount: ->
+    if @props.preferences?.preferences?.grid?
+      @clearRow()
+      @activateTemplate 'grid'
+
+  componentWillUnmount: ->
+    rowID = false
+    @props.annotation.value.map (mark) ->
+      rowID = true if mark._rowID
+    if rowID is true
+      @rowMap '_rowID'
+    else
+      @rowMap 'templateID'
+
+  rowMap: (templateType) ->
+    @props.annotation.value.sort (a,b) ->
+      parseFloat(a.y) - parseFloat(b.y) || parseFloat(a.x) - parseFloat(b.x)
+    tempID = null
+    column = 'a'
+    row = 1
+    for cell in @props.annotation.value
+      if cell[templateType]
+        tempID = cell[templateType] if tempID is null
+        if cell[templateType] == tempID
+          cell.column = column
+          cell.row = row
+          column = String.fromCharCode(column.charCodeAt(0)+1)
+        else
+          row = row + 1
+          tempID = cell[templateType]
+          cell.column = 'a'
+          cell.row = row
+          column = 'b'
+
+  activateTemplate: (type) ->
+    @props.preferences.preferences.activeTemplate = type
+    @props.preferences.update 'preferences'
+    if type is 'grid'
+      @setState gridSelect: true, templateForm: null, error: null
+    if type is null
+      @setState gridSelect: false
+
+  clearRow: ->
+    @removeMarks 'row'
+    @activateTemplate null
+    @props.preferences.preferences.row = null
+    @props.preferences.update 'preferences'
+
+  removeMarks: (type) ->
+    marks = @props.annotation.value.filter((mark) =>
+      mark if mark._type != type
+    )
+    @props.annotation.value = marks
+    @props.preferences.update 'preferences'
+
+  deleteGrid: ->
+    @props.user.get('project_preferences', {project_id: @props.workflow.links.project}).then ([pref]) =>
```

Comment:
They are, but I think I decided to make an API call here because `@props.preferences` wasn't being updated correctly when I tried to alter `@props.preferences` directly. I can try to switch it back and see how it goes. I remember this was a frustrating part with dealing with the preferences and for some reason the grid wasn't behaving.